pglock/core.py

```diff
@@ -222,14 +240,14 @@ class advisory(contextlib.ContextDecorator):
 
     def __init__(
         self,
-        lock_id=None,
+        lock_id: int | str | None = None,
         *,
-        shared=False,
-        xact=False,
-        using=DEFAULT_DB_ALIAS,
-        timeout=_unset,
-        side_effect=None,
-    ):
+        shared: bool = False,
+        xact: bool = False,
+        using: str = DEFAULT_DB_ALIAS,
+        timeout: float | dt.timedelta | _Unset | None = _unset,
+        side_effect: type[SideEffect] | None = None,
```

Comment:
In some cases an actual instance works, but there might a subtle bug in that case so making the type a bit stricter.